context.go

```diff
@@ -0,0 +1,39 @@
+package rpc
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+)
+
+// CtxRpcKey is a type defining the context key for the RPC context
+type CtxRpcKey int
+
+const (
+	// CtxRpcTagsKey defines a context key that can hold a slice of context keys
+	CtxRpcTagsKey CtxRpcKey = iota
+)
+
+type CtxRpcTags map[string]interface{}
+
+// NewContext returns a new Context that carries adds the given log
+// tag mappings (context key -> display string).
+func NewContextWithLogTags(ctx context.Context, logTagsToAdd CtxRpcTags) (context.Context, error) {
+	currTags, ok := RpcTagsFromContext(ctx)
+	if !ok {
```

Comment:
Hmm, couple of questions here. Do we want to modify the existing map if it exists? Or create a deep copy? Or return an error if the value already exists? If we want to modify the existing map, is it necessary to return a new context.WithValue (since the old one will still do)?